port/linux/ipadapter.c

```diff
@@ -969,11 +1041,18 @@ network_event_thread(void *data)
              (unsigned)tv.tv_usec);
     }
 #endif /* OC_HAS_FEATURE_TCP_ASYNC_CONNECT */
-    int n = select(FD_SETSIZE, &rdfds, wfds, NULL, timeout);
 
-    if (FD_ISSET(dev->shutdown_pipe[0], &rdfds)) {
-      process_shutdown(dev);
+#ifdef OC_DYNAMIC_ALLOCATION    
+    if (oc_get_network_events_queue_length(dev->device) >= OC_MAX_NUM_CONCURRENT_REQUESTS) {
+      // the queue is full -> add only control flow rfds
+      FD_ZERO(&rdfds);
+      add_control_flow_rfds(&rdfds, dev);
```

Comment:
extend `ip_context_rfds_fd_copy, tcp_context_cfds_fd_copy, add_control_flow_rfds, tcp_add_controlflow_socks_to_rfd_set` with aditional parameter `int* max_fd ` and use in select(max_fd+1...) and later.

Reply:
Done, added `max_fd ` parameter to improve the performance of functions manipulating with `fd_set` (mostly iterating through all values in the set). `max_fd `value is calculated only when the device file description set changes.